tools/release/backport_pr.sh

```diff
@@ -24,59 +24,100 @@ ensure_command () {
   fi
 }
 
+display_usage () {
+  echo "USAGE: $0 PR_NUMBER GITHUB_USER BACKPORT_BRANCHES REVIEWERS [-c PER_BACKPORT_COMMAND]" >/dev/stderr
+  echo "   PR_NUMBER: The number for the PR to be backported." >/dev/stderr
+  echo "   GITHUB_USER: Your GitHub username." >/dev/stderr
+  echo "   BACKPORT_BRANCHES: A space-separated list of branches to which to backport." >/dev/stderr
+  echo "   REVIEWERS: A comma-separated list of users add as reviewers and assignees." >/dev/stderr
+  echo "   PER_BACKPORT_COMMAND : An optional command to run after cherrypicking the PR to the target branch." >/dev/stderr
+  echo "     If you use this option, ensure your working directory is clean, as `git add -A` will be used to " >/dev/stderr
+  echo "     incorporate any generated files. Try running `git clean -xdff` beforehand." >/dev/stderr
+  echo "" >/dev/stderr
+  echo "Example: $0 25456 gnossen \"v1.30.x v1.31.x v1.32.x v1.33.x v1.34.x v1.35.x v1.36.x\" \"menghanl,gnossen\"" >/dev/stderr
+  echo "Example: $0 25493 gnossen \"\$(seq 30 33 | xargs -n1 printf \"v1.%s.x \")\" \"menghanl\" -c ./tools/dockerfile/push_testing_images.sh" >/dev/stderr
+  exit 1
+}
+
 ensure_command "curl"
 ensure_command "egrep"
 ensure_command "hub"
 ensure_command "jq"
 
+if [ "$#" -lt "4" ]; then
+  display_usage
+fi
+
+PR_NUMBER="$1"
+GITHUB_USER="$2"
+BACKPORT_BRANCHES="$3"
+REVIEWERS="$4"
+shift 4
+
+PER_BACKPORT_COMMAND=""
+while getopts "c:" OPT; do
+  case "$OPT" in
+    c )
+      PER_BACKPORT_COMMAND="$OPTARG"
+      ;;
+    \? )
+      echo "Invalid option: $OPTARG" >/dev/stderr
+      display_usage
+      ;;
+    : )
+      echo "Invalid option: $OPTARG requires an argument." >/dev/stderr
+      display_usage
+      ;;
+  esac
+done
+
+if [[ ! -z "$(git status --porcelain)" && ! -z "$PER_BACKPORT_COMMAND" ]]; then
```

Comment:
I'm not sure I understand the logic here. What is the expected behavior of the script if the working dir is not clean, and the user _did not_ provide a per-backport command?

Reply:
> What is the expected behavior of the script if the working dir is not clean, and the user did not provide a per-backport command?

In that case, the script should emit no error and continue without requiring a clean working tree. I just now verified that this works as intended in this case.